mlflow/store/artifact/databricks_logged_model_artifact_repo.py

```diff
@@ -0,0 +1,156 @@
+            self.files_api.upload(self.path(artifact_path), f, overwrite=True)
+
+    def log_artifact(self, local_file: str, artifact_path: Optional[str] = None) -> None:
+        try:
+            self._log_artifact(local_file, artifact_path)
+        except Exception:
+            _logger.info(
+                (
+                    "Failed to log artifact using Databricks SDK, falling back to "
+                    "DatabricksArtifactRepository"
+                ),
+                exc_info=True,
+            )
+            self.databricks_artifact_repo.log_artifact(local_file, artifact_path)
+            return self.databricks_artifact_repo.log_artifact(local_file, artifact_path)
+
+    def _log_artifacts(self, local_dir: str, artifact_path: Optional[str] = None) -> None:
+        futures = []
+        with ThreadPoolExecutor() as executor:
+            for f in Path(local_dir).rglob("*"):
+                if f.is_file():
+                    fut = executor.submit(self.log_artifact, f, artifact_path)
+                    futures.append(fut)
+
+        for fut in futures:
+            fut.result()
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        try:
+            self._log_artifacts(local_dir, artifact_path)
+        except Exception:
+            _logger.info(
+                (
+                    "Failed to log artifacts using Databricks SDK, falling back to "
+                    "DatabricksArtifactRepository"
+                ),
+                exc_info=True,
+            )
+            self.databricks_artifact_repo.log_artifacts(local_dir, artifact_path)
+
+    def _list_artifacts(self, path: Optional[str] = None) -> list[FileInfo]:
+        file_infos: list[FileInfo] = []
+
+        dest_path = self.path(path)
+        if not self._is_dir(dest_path):
+            return file_infos
+
+        for directory_entry in self.files_api.list_directory_contents(dest_path):
+            relative_path = posixpath.relpath(directory_entry.path, self.root_path)
+            file_infos.append(
+                FileInfo(
+                    path=relative_path,
+                    is_dir=directory_entry.is_directory,
+                    file_size=directory_entry.file_size,
+                )
+            )
+
+        return sorted(file_infos, key=lambda f: f.path)
+
+    def _list_artifacts(self, path: Optional[str] = None) -> list[FileInfo]:
+        try:
+            return self._list_artifacts(path)
+        except Exception:
+            _logger.info(
+                (
+                    "Failed to list artifacts using Databricks SDK, falling back to "
+                    "DatabricksArtifactRepository"
+                ),
+                exc_info=True,
+            )
+            return self.databricks_artifact_repo.list_artifacts(path)
+
+    def __download_file(self, remote_file_path: str, local_path: str) -> None:
+        download_resp = self.files_api.download(self.path(remote_file_path))
```

Comment:
There's a duplicate definition of `_list_artifacts` method which would cause recursion issues. It seems like this should have a different name or the other implementation should be removed.